src/main/java/Victor/Storage/Storage.java

```diff
@@ -0,0 +1,121 @@
+package victor.storage;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+import victor.tasktype.Deadline;
+import victor.tasktype.Event;
+import victor.tasktype.Task;
+import victor.tasktype.Todo;
+
+
+/**
+ * The Storage class contains the method to load data from
+ * a data file and store new data into said data file.
+ *
+ * @author Dominic Fu Ming Jun
+ */
+public class Storage {
+
+    /**
+     * The dataFile variable is a File that is used to store the data file.
+     */
+    private File dataFile;
+    /**
+     * The filePath variable is a String that contains the path to the data file location.
+     */
+    private String filePath;
+
+    /**
+     * The Storage Constructor will take in the filePath of the
+     * data file to retrieve the data from. It would update the
+     * filePath and create a new File using the filePath.
+     *
+     * @param filePath The file path of the data file
+     */
+    public Storage(String filePath) {
+        this.dataFile = new File(filePath);
+        this.filePath = filePath;
+    }
+
+    /**
+     * Loads all the data from the dataFile and use it to fill up an empty ArrayList.
+     * If there is no dataFile since they cannot find it at the filePath location,
+     * then the load() will create a new Victor.txt datafile at said location.
+     * It will then return an empty ArrayList if there is no datafile found.
+     *
+     * @return an ArrayList of tasks.
+     * @throws FileNotFoundException If the file doesn't exist at the file path location
+     * @throws IOException           If the program is unable to create a new file
+     *                               at the file path location
+     */
+    public ArrayList<Task> load() {
+        ArrayList<Task> currentLists = new ArrayList<>();
+        try {
+            Scanner s = new Scanner(dataFile);
+            while (s.hasNext()) {
+                String nextLine = s.nextLine();
+                String[] inputs = nextLine.split("\\|");
+                String fileDataType = inputs[0].trim();
+                switch (fileDataType) {
+                case "T" -> {
+                    Task newTodo = new Todo(inputs[2].trim(), Boolean.parseBoolean(inputs[1].trim()));
+                    currentLists.add(newTodo);
+                }
+                case "D" -> {
+                    Task newDeadline = new Deadline(inputs[2].trim(), Boolean.parseBoolean(inputs[1].trim()),
+                            inputs[3].trim());
+                    currentLists.add(newDeadline);
+                }
+                case "E" -> {
+                    Task newEvent = new Event(inputs[2].trim(), Boolean.parseBoolean(inputs[1].trim()),
+                            inputs[3].trim(), inputs[4].trim());
+                    currentLists.add(newEvent);
+                }
+                default -> {
+                }
```

Comment:
perhaps you could consider the option of abstracting this into a method so what it SLAPs!